### PR TITLE
removed swapcomplete messages from protocol

### LINKED ISSUE #17: BOLD #4 swap protocol changes
https://opendex.network/read/04-swap-protocol - remove `SwapCompleted` message and add hint on when a node is supposed to mark a swap as completed locally.

@erkarl @sangaman

### PR DESCRIPTION
**Pull request by hatmer**
closes https://github.com/opendexnetwork/opendex/issues/17

### CODE REVIEW

File: 04-swap-protocol.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Once an order match is found in the taker’s order book, the swap protocol shou
 5. Taker starts the swap by dispatching the first-leg HTLCs on the DAI payment channel to the maker end, using `r_hash`
 6. Maker is listening for an incoming HTLC on the DAI payment channel. Once it arrives he verifies price and quantity and then dispatches the second-leg HTLCs on the BTC payment channel to the taker end.
 7. Taker is listening for an incoming HTLC on the BTC payment channel. Once it arrives he releases `r_preimage`. This allows **both** the taker and the maker payments to finalize.
-9. Taker sends the `SwapCompleted` message to the maker
+9. Both nodes locally mark the swap as completed.
```
Comment (kilrau) [inline 04-swap-protocol.md:14]:
```suggestion
9. Both nodes locally mark the swap as completed once the respective HTLC was resolved and payment finalized.
```

**Review by kilrau — APPROVED** [review]

File: 04-swap-protocol.md (hunk 1 above)
Comment (kilrau) [inline 04-swap-protocol.md]:
```suggestion
8. Both nodes locally mark the swap as completed once the respective HTLC was resolved and payment finalized.
```

**Review by kilrau — CHANGES_REQUESTED** [review]

**Comment by kilrau** [comment]
This has been wanted for long: getting rid of the redundant 8th `SwapCompleted`. Since we ran into issues with that when integrating another swap client, we are tackling this now.

**Comment by kilrau** [comment]
@LePremierHomme @erkarl @sangaman

**Review by ghost — APPROVED** [review]

**Comment by hatmer** [comment]
Makes sense, new version is up with a few extra grammar edits

File: 04-swap-protocol.md [hunk 2]
```diff
@@ -9,9 +9,9 @@ Once an order match is found in the taker’s order book, the swap protocol shou
...
 4. Maker confirms full or partial quantity in the `SwapAccepted` message
 5. Taker starts the swap by dispatching the first-leg HTLCs on the DAI payment channel to the maker end, using `r_hash`
-6. Maker is listening for an incoming HTLC on the DAI payment channel. Once it arrives he verifies price and quantity and then dispatches the second-leg HTLCs on the BTC payment channel to the taker end.
-7. Taker is listening for an incoming HTLC on the BTC payment channel. Once it arrives he releases `r_preimage`. This allows **both** the taker and the maker payments to finalize.
-9. Taker sends the `SwapCompleted` message to the maker
+6. Maker listens for an incoming HTLC on the DAI payment channel. Once it arrives he verifies price and quantity and then dispatches the second-leg HTLCs on the BTC payment channel to the taker end.
+7. Taker listens for an incoming HTLC on the BTC payment channel. Once it arrives he releases `r_preimage`. This allows **both** the taker and the maker payments to finalize.
+9. Both nodes locally mark the swap as completed once the respective HTLC is resolved and the payment is finalized.
```
Comment (kilrau) [inline 04-swap-protocol.md]:
```suggestion
8. Both nodes locally mark the swap as completed once the respective HTLC is resolved and the payment is finalized.
```

**Review by kilrau — CHANGES_REQUESTED** [review]

File: 04-swap-protocol.md (hunk 2 above)
Comment (kilrau) [reply 04-swap-protocol.md]: GitBook corrects this automatically, but for someone viewing the plain markdown files we want numbering to be correct.

**Review by kilrau — APPROVED** [review]

**Pull request merged**